### PR TITLE
Publisher forms

### PR DESCRIPTION
**Pull request by acabunoc**
Publisher form work

Functionality -- send emails to authors notifying them they can claim badges. Generate id for unique url.

### CODE REVIEW

**Comment by acabunoc** [comment]
@alicoding ready for review! Including publisher security for creating papers. I'm around all afternoon if you have questions :)

File: src/routes/claims/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  var query  = Claim.where({ slug: request.params.slug });
+  query.findOne(function(err, claim) {
+    response.json(claim);
+  });
+}
+
+module.exports = function () {
+  return {
+    // GET
+    getClaim: getClaim
```
Comment (acabunoc) [inline src/routes/claims/index.js]: rewrite to just export the one function! (Ali) -- & make it a singleton

File: templates/client.jsx [hunk 2]
```diff
@@ -8,6 +9,7 @@ var routes = (
     <Route name="home" path="/" handler={require('./pages/home.jsx')} />
     <Route name="about" path="/about/?" handler={require('./pages/about.jsx')} />
     <Route name="issue" path="/issue/?" handler={require('./pages/issue.jsx')} />
+    <Route name="issue-slug" path="/issue/:slug/?" handler={require('./pages/issue.jsx')} />
```
Comment (acabunoc) [inline templates/client.jsx]: Add a '?' to make :slug optional

File: templates/client.jsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 var React = require('react'),
     Router = require('react-router'),
+    fetch = require('isomorphic-fetch'),
```
Comment (acabunoc) [inline templates/client.jsx]: add to webpack? http://mts.io/2015/04/08/webpack-shims-polyfills/

File: templates/pages/issue.jsx [hunk 4]
```diff
@@ -6,18 +6,43 @@ var React = require('react'),
...
+    router: React.PropTypes.func
+  },
   componentWillMount: function() {
     document.title = "Contributorship Badges";
-    if(!this.props.orcid){
+    if(!this.props.user){
       //redirect if user isn't logged in
       window.location.href="/request-orcid-user-auth";
     }
+    this.loadClaimFromServer(this.context.router.getCurrentParams().slug);
```
Comment (acabunoc) [inline templates/pages/issue.jsx]: with react-router-- look up this.params.slug

File: templates/pages/paper.jsx [hunk 5]
```diff
@@ -1,26 +1,39 @@
...
   componentWillMount: function() {
     document.title = "Submit a Paper | Contributorship Badges";
+    if(!this.props.user){
+      //redirect if user isn't logged in
+      window.location.href="/request-orcid-user-auth";
+    }
+
+    if(this.props.user.role != 'publisher'){
+      //redirect home is user isn't a publisher
+      window.location.href="/";
```
Comment (acabunoc) [inline templates/pages/paper.jsx]: https://rackt.github.io/react-router/#Custom%20Location

replacewith

File: templates/pages/paper.jsx [hunk 6]
```diff
@@ -29,7 +42,7 @@ var Issue = React.createClass({
         return response.json();
     })
     .then((data) => {
-        document.location = url + '/badges';
+        window.location.reload();
```
Comment (acabunoc) [inline templates/pages/paper.jsx]: clear state instead

**Comment by acabunoc** [comment]
@alicoding finished updates! Let me know if you want to take a last pass :)

File: templates/pages/issue.jsx (hunk 4 above)
Comment (acabunoc) [inline templates/pages/issue.jsx]: couldn't find it :(

**Comment by alicoding** [comment]
r+! \o/

**Pull request merged**